File: Insertionsort/cpp/insertionCharCrescente.cpp
#include <iostream>

/* 
Descricao: essa funcao ordena um vetor de caracteres
em ordem alfabetica com o metodo insertionsort

Parametro: um vetor de caracteres (vetor a ser ordenado)
e um inteiro (tamanho do vetor)
*/
void insertionsort (char vet[], int n) {
	for (int i = 1; i < n; i++) {
		char temp = vet[i];
		int j = i - 1;

		while (j>=0 && vet[j] > temp) {
			vet[j+1] = vet[j];
			j--;
		}

		vet[j+1] = temp;
	}
}

File: Insertionsort/java/insertionCharCrescente.java
public class insertionCharCrescente {
	/* 
	* Descricao: essa funcao ordena um vetor de caracteres
	* em ordem alfabetica com o metodo insertionsort
	* 
	* Parametro: um vetor de caracteres (vetor a ser ordenado)
	*/
	public static void insertionsort (char[] vet) {
		for (int i = 1; i < vet.length; i++) {
			char temp = vet[i];
			int j = i - 1;

			while (j>=0 && vet[j] > temp) {
				vet[j+1] = vet[j];
				j--;
			}

			vet[j+1] = temp;
		}
	}
}
